bikeindex/bike_index
# Anonymize db

## Overview

This PR adds **postgres comment**s to all columns that should be **anonymized**  when the database is dumped via `pg_dump`.  The `fake_pipe` library will reflect on all pg comments when the database is dumped and will anonymize the data with `Faker` for the respective target type, e.g, `sentence`, `word`, `bcrypt_password`, `md5`, etc. ([supported conversions](https://github.com/gadogado/fake_pipe#currently-support-mutation-methods))

## Changes
There's a new migration that adds all of the comments.  This needs to be run on the database before its dumped and anonymized.  After this is complete you can use a new rake task to dump, restore, and even setup an anon user for testing.

## How to test
1.  Load your database ( or use the existing one ) with whatever db you want to anonymize.
2. Run the migrations
3. Dump and and anonymize the output file: `rake db:anon_dump` ( Takes ~ 8 mins on my local )
4. Restore with the anon dump file that was created: `rake db:anon_restore`
5. Setup one of the users for dev testing: `rake db:setup_anon_user`
6. Do a manual audit and make sure everything is anonymized.  I usually use something like [postico](https://eggerapps.at/postico/) for this
## Possible issues
1.  The dump process should ignore acls and owners.  If there's any issues you can create a missing user with the usual: `CREATE USER postgres SUPERUSER;`

## TODOS

- [x] fork fake_pipe to add a number mutator
